In [3]:
import pandas as pd

## pandas  series 
-A series is a id array . this means it contains a singe column

In [8]:
## creating a series data
student_scores = pd.Series(
    [89, 98, 84, 67, 56],
    name="student scores",
    index=["Imran", "Kato", "Anderw", "Wassawa", "John"]
)
print(student_scores)

Imran      89
Kato       98
Anderw     84
Wassawa    67
John       56
Name: student scores, dtype: int64


## 1(a)

In [2]:
student_data={
    "Name":["Imran", "Kato", "Anderw", "Wassawa", "John"],
    "Maths":[89, 98, 84, 67, 56],
    "sic":[89, 98, 84, 67, 56],
    "gender":["male", "female", "male", "male", "female",],
}
students = pd.DataFrame(student_data)
students

,Name,Maths,sic,gender
0,Imran,89,89,male
1,Kato,98,98,female
2,Anderw,84,84,male
3,Wassawa,67,67,male
4,John,56,56,female


In [15]:
students.tail(0)## bottom
              

,Name,Maths,sic,gender


## MEASERS OF CENTRAL TENDENCY 
THE are only three for example 
- Mean 
- median
- Mode 

In [1]:
import pandas as pd

In [3]:
student_data={
    "Name":["Imran", "Kato", "Anderw", "Wassawa", "John"],
    "Maths":[89, 98, 84, 67, 56],
    "Science":[89, 98, 84, 67, 56],
    "gender":["male", "female", "male", "male", "female",],
}
students = pd.DataFrame(student_data)
students

,Name,Maths,Science,gender
0,Imran,89,89,male
1,Kato,98,98,female
2,Anderw,84,84,male
3,Wassawa,67,67,male
4,John,56,56,female


In [5]:
students["Maths"]

0    89
1    98
2    84
3    67
4    56
Name: Maths, dtype: int64

## ;MEAN
Finding the mean (findng the mean maths mark)

In [8]:
students["Maths"].mean()


np.float64(78.8)

In [9]:
students["Maths"].median()

np.float64(84.0)

In [15]:
students["Maths"].mode()[0]
mode = students["Maths"].mode()  
print("Mode of Maths scores:", mode[0])

Mode of Maths scores: 56


## Find the mean Science mark and explain your results 
## Find the median of the Science mark and list the students attached to it 

In [17]:
students['Science'].mean()

np.float64(78.8)

after adding all the marks scored by all students and dividing by the tatol number of students i got my mean as (78.8)


In [ ]:
students["science"].median()

## MEASURES diSIPATION
- STD
- VAR
- RANGE

## QN
- FIND THE SPREAD FOR SIC MARKS USING V
- WAT IS THE RANG OF THE SINCE MARKS AND COMENT ON U ANSWER

In [20]:
## variance of Science scores
students["Science"].var()

np.float64(289.70000000000005)

In [21]:
## RANGE of Science scores
students["Science"].max() - students["Science"].min()


np.int64(42)

42 is the max and and minium mark for sicence 

## MEASURES OF POSITION
-PERCENTILES 
-QUARTRTILES
-DECILE etc

In [22]:
## QUARTILES 
### lower  quantile (Q1)
lower_quantile =students["Science"].quantile(0.25)
lower_quantile 

np.float64(67.0)

## PERCNTILES 

In [ ]:
import numpy as np

